### PR TITLE
Add itest coverage for custom channels strict forwarding

### PR DESCRIPTION
**Pull request by GeorgeTsagk**
## Description

This PR adds coverage for the strict forwarding behavior of custom channels.

## Dependent PRs

- [ ] Tapd [#1144](https://github.com/lightninglabs/taproot-assets/pull/1144)
- [x] Lnd [#9199](https://github.com/lightningnetwork/lnd/pull/9199)
- [x] Lndclient [#196](https://github.com/lightninglabs/lndclient/pull/196)

`go.mod` will be updated once the above PRs have been merged, in order to point to them

### CODE REVIEW

File: itest/assets_test.go [hunk 1]
```diff
@@ -727,6 +727,42 @@ func payInvoiceWithSatoshi(t *testing.T, payer *HarnessNode,
...
+	invoice *lnrpc.AddInvoiceResponse, hopPub []byte,
+	expectedStatus lnrpc.Payment_PaymentStatus) {
+
+	ctxb := context.Background()
+	ctxt, cancel := context.WithTimeout(ctxb, defaultTimeout)
+	defer cancel()
+
+	routeRes, err := payer.RouterClient.BuildRoute(
+		ctxb,
+		&routerrpc.BuildRouteRequest{
```
Comment (guggero) [inline itest/assets_test.go:740]: nit: could fit on previous line with `ctxb`.

File: itest/assets_test.go [hunk 2]
```diff
@@ -727,6 +727,42 @@ func payInvoiceWithSatoshi(t *testing.T, payer *HarnessNode,
...
+			AmtMsat:        17800,
+			FinalCltvDelta: 80,
+			PaymentAddr:    invoice.PaymentAddr,
+			HopPubkeys:     [][]byte{hopPub},
+		},
+	)
+	require.NoError(t, err)
+
+	res, err := payer.RouterClient.SendToRouteV2(ctxt,
```
Comment (guggero) [inline itest/assets_test.go:749]: nit: move `ctxt` to next line.

File: itest/assets_test.go [hunk 3]
```diff
@@ -727,6 +727,42 @@ func payInvoiceWithSatoshi(t *testing.T, payer *HarnessNode,
...
+		&routerrpc.SendToRouteRequest{
+			PaymentHash: invoice.RHash,
+			Route:       routeRes.Route,
+		})
```
Comment (guggero) [inline itest/assets_test.go:753]: nit: closing parenthesis on next line to achieve symmetry.

File: itest/litd_custom_channels_test.go [hunk 4]
```diff
@@ -1943,4 +1943,21 @@ func testCustomChannelsLiquidityEdgeCases(_ context.Context,
...
+	)
+
+	// Yara pays Dave with enough satoshis, but Charlie will not settle as
+	// he expects assets.
+	payInvoiceWithSatoshiLastHop(
+		t.t, yara, invoiceResp, dave.PubKey[:], lnrpc.Payment_FAILED,
+	)
+
+	logBalance(t.t, nodes, assetID, "after failed payment (asset "+
+		"invoice, strict forwarding)")
```
Comment (guggero) [inline itest/litd_custom_channels_test.go:1962]: We could also assert that the invoice is still open:

```
	ctxb := context.Background()
	stream, err := dave.InvoicesClient.SubscribeSingleInvoice(
		ctxb, &invoicesrpc.SubscribeSingleInvoiceRequest{
			RHash: invoiceResp.RHash,
		},
	)
	require.NoError(t.t, err)

	// Yara pays Dave with enough satoshis, but Charlie will not settle as
	// he expects assets.
	payInvoiceWithSatoshiLastHop(
		t.t, yara, invoiceResp, dave.PubKey[:], lnrpc.Payment_FAILED,
	)

	t.lndHarness.LNDHarness.AssertInvoiceState(stream, lnrpc.Invoice_OPEN)

	logBalance(t.t, nodes, assetID, "after failed payment (asset "+
		"invoice, strict forwarding)")
```

**Review by guggero — COMMENTED** [review]
Nice, LGTM apart from some nits.

Not approving yet because of dependencies.

**Comment by GeorgeTsagk** [comment]
Updated to point to latest state of tapd/lnd branches, before we change their base and things break:
- tapd: https://github.com/lightninglabs/taproot-assets/pull/1144/commits/3a80ed512ee5458d24c7506be533b91ea1ad3343
- lnd: https://github.com/lightningnetwork/lnd/pull/9199/commits/d38204e3fe5897bdfeb89b2101768026b9dc8211

**Comment by guggero** [comment]
Removing request for review until dependencies are in.

**Review by jharveyb — APPROVED** [review]
LGTM, good test!

**Pull request merged**